base/loading.jl

```diff
@@ -80,25 +80,331 @@ else
     end
 end
 
-macro return_if_file(path)
-    quote
-        path = $(esc(path))
-        isfile_casesensitive(path) && return path
+## SHA1 ##
+
+struct SHA1
+    bytes::Vector{UInt8}
+    function SHA1(bytes::Vector{UInt8})
+        length(bytes) == 20 ||
+            throw(ArgumentError("wrong number of bytes for SHA1 hash: $(length(bytes))"))
+        return new(bytes)
     end
 end
+SHA1(s::Union{String,SubString{String}}) = SHA1(hex2bytes(s))
```

Comment:
This is just internal, we should not export this type or anything and should maybe replace it with something else, but for now it can live here and we can clean this stuff up later.